src/System.Net.Http/tests/UnitTests/HttpSystemProxyTest.cs

```diff
@@ -33,12 +33,50 @@ public void HttpProxy_SystemProxy_Loaded()
             Assert.False(HttpSystemProxy.TryCreate(out p));
 
             FakeRegistry.WinInetProxySettings.Proxy = FakeProxyString;
-            WinInetProxyHelper proxyHelper = new WinInetProxyHelper();
 
             Assert.True(HttpSystemProxy.TryCreate(out p));
             Assert.NotNull(p);
             Assert.Equal(fakeProxyUri, p.GetProxy(fooHttp));
             Assert.Equal(fakeProxyUri, p.GetProxy(fooHttps));
         }
+
+        [Theory]
+        [InlineData("http://localhost/", true)]
+        [InlineData("http://127.0.0.1/", true)]
+        [InlineData("http://128.0.0.1/", false)]
+        [InlineData("http://[::1]/", true)]
+        [InlineData("http://foo/", true)]
+        [InlineData("http://www.foo.com/", true)]
+        [InlineData("http://WWW.FOO.COM/", true)]
+        [InlineData("http://foo.com/", false)]
+        [InlineData("http://bar.com/", true)]
+        [InlineData("http://BAR.COM/", true)]
+        [InlineData("http://162.1.1.1/", true)]
+        [InlineData("http://[2a01:5b40:0:248::52]/", false)]
+        [InlineData("http://[2002::11]/", true)]
+        [InlineData("http://[2607:f8b0:4005:80a::200e]/", true)]
+        [InlineData("http://[2607:f8B0:4005:80A::200E]/", true)]
+        [InlineData("http://b\u00e9b\u00e9.eu/", true)]
+        [InlineData("http://www.b\u00e9b\u00e9.eu/", true)]
+        public void HttpProxy_Local_Bypassed(string name, bool shouldBypass)
+        {
+            RemoteInvoke((url, expected) =>
+            {
+                bool expectedResult = Boolean.Parse(expected);
+                IWebProxy p;
+
+                FakeRegistry.Reset();
+                FakeRegistry.WinInetProxySettings.Proxy = FakeProxyString;
+                FakeRegistry.WinInetProxySettings.ProxyBypass = "23.23.86.44;*.foo.com;<local>;BAR.COM; ; 162*;[2002::11];[*:f8b0:4005:80a::200e]; http://www.xn--mnchhausen-9db.at;http://*.xn--bb-bjab.eu;http://xn--bb-bjab.eu;";
```

Comment:
It'd be good to validate other ProxyBypass entries, such as one that's empty, one that contains a single address, one that contains multiple colons in a row, etc.  Basically make sure we have tests for all the various special cases and conditions tested for in the product code.